### PR TITLE
PISTON-651: don't save as retry for delete_docs conflict

### PR DESCRIPTION
**Pull request by danielfinke**
(when doing an unassign_from_prev during a save of a number in a new
account)

### CODE REVIEW

**Comment by k-anderson** [comment]
@icehess @lazedo could one of you test / review this PR please?  Been here a bit and needs comments or acceptance ;)

Thanks!

File: core/kazoo_number_manager/src/knm_phone_number.erl [hunk 1]
```diff
@@ -256,27 +268,38 @@ handle_bulk_change(Db, JObjs, PNsMap, T0, ErrorF)
...
+    RetryF = fun(Db1, PNs1, Num, T1) ->
+                     retry_save(Db1, PNs1, ErrorF, Num, T1)
+             end,
+    handle_bulk_change(Db, JObjs, PNs, T, ErrorF, RetryF).
+
+-spec retry_save(kz_term:ne_binary(), pns_map(), bulk_change_error_fun(), knm_numbers:num(), knm_numbers:collection()) ->
+                        knm_numbers:collection().
+retry_save(Db, PNsMap, ErrorF, Num, T) ->
+    PN = maps:get(Num, PNsMap),
+    case kz_datamgr:ensure_saved(Db, to_json(PN)) of
```
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/knm_phone_number.erl]: Hi @danielfinke sorry this has languished so long. Can you remove the use of the deprecated `ensure_saved` in favor of `update_doc`? Then I will merge this

Comment (danielfinke) [reply core/kazoo_number_manager/src/knm_phone_number.erl]: @jamesaimonetti gonna switch it to `save_doc/2` instead unless there's a reason I'm missing as to why you suggested `update_doc`?

Comment (jamesaimonetti) [reply core/kazoo_number_manager/src/knm_phone_number.erl]: Only that update_doc has an ensure_saved flag

Comment (danielfinke) [reply core/kazoo_number_manager/src/knm_phone_number.erl]: @jamesaimonetti ok, using update_doc/3 instead

Comment (danielfinke) [reply core/kazoo_number_manager/src/knm_phone_number.erl]: @jamesaimonetti hey, anything else needed from your end to get this thru?

**Comment by jamesaimonetti** [comment]
Hey @danielfinke can you rebase to get CI to run again?

**Comment by danielfinke** [comment]
@jamesaimonetti done :)

**Comment by jamesaimonetti** [comment]
Merged!

**Pull request closed without merging**